backend/Squirrel.Core/Squirrel.Core.BLL/Services/UserService.cs

```diff
@@ -54,7 +71,82 @@ public async Task<User> CreateUserAsync(UserRegisterDto userDto, bool isGoogleAu
         var createdUser = (await _context.Users.AddAsync(newUser)).Entity;
         await _context.SaveChangesAsync();
 
-        return createdUser;
+        return _mapper.Map<UserDto>(createdUser); ;
+    }
+
+    public async Task<UserProfileDto> UpdateUserNamesAsync(UpdateUserNamesDto updateUserDTO)
+    {
+        var userEntity = await GetUserByIdInternal(updateUserDTO.Id);
+
+        if (await GetUserByUsernameInternal(updateUserDTO.Username) is not null)
+        {
+            throw new UsernameAlreadyRegisteredException();
```

Comment:
what this check is used for?
I just try to change FirstName and received error
![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/70843124/42ecea36-4e92-4202-a32f-7d93c16ffa6f)